source/array-index.d.ts

```diff
@@ -0,0 +1,20 @@
+/**
+Extract the keys of an array or tuple.
+
+Use-case: Deal with valid keys for a const array.
+
+@example
+```
+import type {ArrayIndex} from 'type-fest';
+
+const weekDays = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday' ] as const;
+
+type WeekDay = ArrayIndex<typeof weekDays>;
+
+const getWeekDayName = (day: WeekDay): WeekDayName => weekDays[day];
```

Comment:
This is missing the `WeekDayName` type, and the example is kinda incomplete without a `ArrayValue` type.

Reply:
Agree, the PR for `ArrayValue` looks stoled, I can create another for it to merge together